### PR TITLE
docs: COPY INTO updates

### PR DESCRIPTION
**Pull request by soyeric128**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- added examples:  Loading to Table with Extra Columns.
- added examples: Filtering Files with Pattern.
- other refactors to make the page clean and organized.

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13003)

### CODE REVIEW

File: docs/doc/12-load-data/00-transform/07-data-load-transform.md [hunk 1]
```diff
@@ -2,7 +2,23 @@
...
+     FROM ( SELECT [<file_col> ... ]
+            FROM { userStage | internalStage | externalStage } )
+[ FILES = ( '<file_name>' [ , '<file_name>' ] [ , ... ] ) ]
+[ PATTERN = '<regex_pattern>' ]
+[ FILE_FORMAT = ( TYPE = {PARQUET} [ formatTypeOptions ] ) ]
+[ copyOptions ]
+```
+
+:::note
+This feature is currently only available for the Parquet file format.
```
Comment (BohuTANG) [inline docs/doc/12-load-data/00-transform/07-data-load-transform.md:18]: we support CSV/Parquet both with: `select $1, $2 ...`

Comment (BohuTANG) [reply docs/doc/12-load-data/00-transform/07-data-load-transform.md:18]: https://databend.rs/doc/load-data/transform/querying-stage

File: docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md [hunk 2]
```diff
@@ -221,81 +214,103 @@ copyOptions ::=
...
 | FILE             | VARCHAR  | NO       | The relative path to the source file.       |
 | ROWS_LOADED      | INT      | NO       | The number of rows loaded from the source file. |
 | ERRORS_SEEN      | INT      | NO       | Number of error rows in the source file    |
 | FIRST_ERROR      | VARCHAR  | YES      | The first error found in the source file.             |
 | FIRST_ERROR_LINE | INT      | YES      | Line number of the first error.             |
 
+## Managing Parallel Processing
 
-## Examples
+In Databend, the `max_threads` setting specifies the maximum number of threads that can be utilized to execute a request. By default, this value is typically set to match the number of CPU cores available on the machine.
```
Comment (BohuTANG) [inline docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md:231]: It's better not let the user know the settings like `max_threads`, all is done internal for them.

File: docs/doc/12-load-data/00-transform/07-data-load-transform.md (hunk 1 above)
Comment (youngsofun) [reply docs/doc/12-load-data/00-transform/07-data-load-transform.md:18]: yes all format support thansform, the select part is the same as  https://databend.rs/doc/load-data/transform/querying-stage

File: docs/doc/12-load-data/00-transform/07-data-load-transform.md [hunk 3]
```diff
@@ -2,7 +2,23 @@
...
-Databend offers a powerful feature that enables data transformation during the loading process using the [COPY INTO](/14-sql-commands/10-dml/dml-copy-into-table.md) command. This functionality simplifies your ETL pipeline by incorporating basic transformations, eliminating the need for temporary tables. By transforming data during loading, you can streamline your ETL process effectively. Here are practical ways to enhance data loading with this feature:
+Databend offers a powerful feature that enables data transformation during the loading process using the [COPY INTO](/14-sql-commands/10-dml/dml-copy-into-table.md) command, with this syntax:
+
+```sql
+COPY INTO [<database>.]<table_name> [ ( <col_name> [ , <col_name> ... ] ) ]
+     FROM ( SELECT [<file_col> ... ]
+            FROM { userStage | internalStage | externalStage } )
+[ FILES = ( '<file_name>' [ , '<file_name>' ] [ , ... ] ) ]
+[ PATTERN = '<regex_pattern>' ]
+[ FILE_FORMAT = ( TYPE = {PARQUET} [ formatTypeOptions ] ) ]
```
Comment (youngsofun) [inline docs/doc/12-load-data/00-transform/07-data-load-transform.md:13]: the `PARQUET` here not needed too

File: docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md [hunk 4]
```diff
@@ -332,81 +338,114 @@ CONNECTION = (endpoint_url = 'https://<your-ipfs-gateway>')
...
-PATTERN = '.*sales.*[.]csv'
-FILE_FORMAT = (type = CSV field_delimiter = ',' record_delimiter = '\n' skip_header = 1);
+FROM 's3://mybucket/data.csv.gz'
+CONNECTION = (
+    ENDPOINT_URL = 'https://<endpoint-URL>',
+    ACCESS_KEY_ID = '<your-access-key-ID>',
+    SECRET_ACCESS_KEY = '<your-secret-access-key>'
+)
+FILE_FORMAT = (type = CSV field_delimiter = ',' record_delimiter = '\n' skip_header = 1 compression = AUTO)
+SIZE_LIMIT = 10;
```
Comment (youngsofun) [inline docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md:356]: the SIZE_LIMIT is wrongly implemented now, please  remove `SIZE_LIMIT = 10;` from example

File: docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md [hunk 5]
```diff
@@ -332,81 +338,114 @@ CONNECTION = (endpoint_url = 'https://<your-ipfs-gateway>')
...
 ```
 </TabItem>
-
 </Tabs>
 
-### 4. Loading Data with Pattern Matching
+### Example 3: Loading Compressed Data
 
-This example uses pattern matching to only load from CSV files containing `sales` in their names:
+This example loads 10 rows of data from a GZIP-compressed CSV file on Amazon S3 into Databend:
```
Comment (youngsofun) [inline docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md:345]: here too

Comment (youngsofun) [inline docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md:345]:
```suggestion
This example loads a GZIP-compressed CSV file on Amazon S3 into Databend:
```

File: docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md (hunk 4 above)
Comment (youngsofun) [inline docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md:356]:
```suggestion
```

File: docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md [hunk 6]
```diff
@@ -332,81 +338,114 @@ CONNECTION = (endpoint_url = 'https://<your-ipfs-gateway>')
...
-FROM 's3://mybucket/'
-PATTERN = '.*sales.*[.]csv'
-FILE_FORMAT = (type = CSV field_delimiter = ',' record_delimiter = '\n' skip_header = 1);
+FROM 's3://mybucket/data.csv.gz'
+CONNECTION = (
+    ENDPOINT_URL = 'https://<endpoint-URL>',
+    ACCESS_KEY_ID = '<your-access-key-ID>',
+    SECRET_ACCESS_KEY = '<your-secret-access-key>'
+)
+FILE_FORMAT = (type = CSV field_delimiter = ',' record_delimiter = '\n' skip_header = 1 compression = AUTO)
```
Comment (youngsofun) [inline docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md:355]:
```suggestion
FILE_FORMAT = (type = CSV field_delimiter = ',' record_delimiter = '\n' skip_header = 1 compression = AUTO);
```

File: docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md (hunk 4 above)
Comment (youngsofun) [inline docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md:356]:
```suggestion
```

**Review by youngsofun — APPROVED** [review]

**Pull request merged**